configs/projects/puppet-agent.rb

```diff
@@ -211,8 +211,8 @@
   if platform.is_huaweios?
     proj.component "rubygem-net-scp"
     proj.component "rubygem-mini_portile2"
-    proj.component "rubygem-nokogiri"
     proj.component "rubygem-net-netconf"
+    proj.component "rubygem-nokogiri"
```

Comment:
In some experimentation I did, the order the components are declared did matter. I'm pretty sure of this, but not 100%.